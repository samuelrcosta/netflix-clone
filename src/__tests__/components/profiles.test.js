import { render, fireEvent } from '@testing-library/react';
import { Profiles } from '../../components';

describe('<Profiles />', () => {
  it('renders the <Profiles /> with populated data', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture src="5" alt="Joe Doe" data-testid="profile-picture" />
            <Profiles.Name>Joe Doe</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId('profile-picture')).toBeTruthy();
    expect(getByText('Joe Doe')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Profiles /> with populated data but misc profile image', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture data-testid="profile-picture-misc" />
            <Profiles.Name>Joe Doe</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId('profile-picture-misc')).toBeTruthy();
    expect(getByText('Joe Doe')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
