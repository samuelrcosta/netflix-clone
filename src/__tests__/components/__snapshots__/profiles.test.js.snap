// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-bxtkkv-0 fGoDNJ"
>
  <h1
    class="profiles__Title-bxtkkv-1 hPuemP"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 boiWzJ"
  >
    <li
      class="profiles__User-bxtkkv-5 evaSaO"
    >
      <img
        alt="Joe Doe"
        class="profiles__Picture-bxtkkv-4 gogdqh"
        data-testid="profile-picture"
        src="/images/users/5.png"
      />
      <p
        class="profiles__Name-bxtkkv-3 jQTvVi"
      >
        Joe Doe
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders the <Profiles /> with populated data but misc profile image 1`] = `
<div
  class="profiles__Container-bxtkkv-0 fGoDNJ"
>
  <h1
    class="profiles__Title-bxtkkv-1 hPuemP"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 boiWzJ"
  >
    <li
      class="profiles__User-bxtkkv-5 evaSaO"
    >
      <img
        class="profiles__Picture-bxtkkv-4 gogdqh"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-bxtkkv-3 jQTvVi"
      >
        Joe Doe
      </p>
    </li>
  </ul>
</div>
`;
